<launch>
    <arg name="fps" value="1"/>
    <arg name="camera_serial" default="41810612"/>
    <arg name="frame_width" default="2448"/>
    <arg name="frame_height" default="2048"/>

    <arg name="exposure_auto" value="false"/>
    <arg name="gain_auto" value="false"/>
    <arg name="exposure" default="10000"/>
    <arg name="gain" default="100"/>

    <arg name="pipeline" value="video/x-raw, format=(string)GRAY8, width=$(arg frame_width), height=$(arg frame_height),framerate=$(arg fps)/1 ! videoconvert --verbose"/>
        
    <!-- Start the GSCAM nodes -->
    <!--<node ns="cam" name="cam_node" pkg="gscam" type="gscam" output="screen">
        <param name="camera_name" value="cam"/>
        <param name="camera_info_url" value="file:///home/i3dr/.ros/camera_info/cam_info.yaml"/>
        <param name="gscam_config" value="tcamsrc serial=$(arg camera_serial) name=source ! $(arg pipeline)"/>
        <param name="frame_id" value="/cam_depth_optical_frame"/>
        <param name="sync_sink" value="true"/>
    </node>-->

    <node ns="cam" name="i3dr_camera_ctrl" pkg="i3dr_stereo_camera" type="tiscamera_ctrl.py" output="screen">
        <param name="Serial" value="$(arg camera_serial)"/>
        <param name="Width" value="$(arg frame_width)"/>
        <param name="Height" value="$(arg frame_height)"/>
        <param name="FPS" value="$(arg fps)"/>
        <param name="Exposure" value="$(arg exposure)"/>
        <param name="Gain" value="$(arg gain)"/>
        <param name="Exposure_Auto" value="$(arg exposure_auto)"/>
        <param name="Gain_Auto" value="$(arg gain_auto)"/>
    </node>

    <!--<node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_controls" />-->
    <node pkg="rqt_gui" type="rqt_gui" name="test" args="-s reconfigure"/>
</launch>